contracts/test-utils/src/mocha_blockchain.ts

```diff
@@ -88,6 +98,71 @@ export class BlockchainTestsEnvironmentSingleton {
     }
 }
 
+/**
+ * `BlockchainTestsEnvironment` that uses a live web3 provider.
+ */
+export class LiveBlockchainTestsEnvironmentSingleton {
+    private static _instance: BlockchainTestsEnvironmentSingleton | undefined;
+
+    public blockchainLifecycle: BlockchainLifecycle;
+    public provider: Web3ProviderEngine;
+    public txDefaults: Partial<TxData>;
+    public web3Wrapper: Web3Wrapper;
+
+    // Create or retrieve the singleton instance of this class.
+    public static create(): LiveBlockchainTestsEnvironmentSingleton {
+        if (LiveBlockchainTestsEnvironmentSingleton._instance === undefined) {
+            LiveBlockchainTestsEnvironmentSingleton._instance = new LiveBlockchainTestsEnvironmentSingleton();
+        }
+        return LiveBlockchainTestsEnvironmentSingleton._instance;
+    }
+
+    protected static _createWeb3Provider(rpcHost: string): Web3ProviderEngine {
+        const providerEngine = new Web3ProviderEngine();
+        providerEngine.addProvider(new RPCSubprovider(rpcHost));
+        providerUtils.startProviderEngine(providerEngine);
+        return providerEngine;
+    }
+
+    // Get the singleton instance of this class.
+    public static getInstance(): LiveBlockchainTestsEnvironmentSingleton | undefined {
+        return LiveBlockchainTestsEnvironmentSingleton._instance;
+    }
+
+    public async getChainIdAsync(): Promise<number> {
+        return providerUtils.getChainIdAsync(this.provider);
+    }
+
+    public async getAccountAddressesAsync(): Promise<string[]> {
+        return this.web3Wrapper.getAvailableAddressesAsync();
+    }
+
+    protected constructor() {
+        const snapshotHandlerAsync = async (): Promise<void> => {
+            throw new Error('Snapshots are not supported with a live provider.');
+        };
+        this.blockchainLifecycle = {
+            startAsync: snapshotHandlerAsync,
+            revertAsync: snapshotHandlerAsync,
+        } as any;
+        this.txDefaults = txDefaults;
+        if (process.env.LIVE_RPC_URL) {
```

Comment:
This provider is instantiated differently than the `FORK_RPC_URL` provider. Maybe we should just consolidate all of that here?